src/plugins/data/server/search/strategies/eql_search/eql_search_strategy.ts

```diff
@@ -27,15 +28,19 @@ export const eqlSearchStrategyProvider = (
   searchConfig: SearchConfigSchema,
   logger: Logger
 ): ISearchStrategy<EqlSearchStrategyRequest, EqlSearchStrategyResponse> => {
-  async function cancelAsyncSearch(id: string, esClient: IScopedClusterClient) {
+  function cancelAsyncSearch(id: string, esClient: IScopedClusterClient) {
     const client = esClient.asCurrentUser.eql;
-    await client.delete({ id });
+    return client.delete({ id });
```

Comment:
3 notes from my end
* 🙏 thx for quickly providing an alternative implementation
* worth to consider here and `ese`, if a async search was already deleted, so it's a 404, do we need to handle it as an error? In this case it has already been deleted, hasn't it?
* Is there a need to backport this PR? What are the trade-offs if we don't?

Reply:
I do agree and would hesitate to backport it for those reasons, unless it's a necessary fix

Reply:
I'm fine with targeting only 8.12, I'll update the labels.